parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java

```diff
@@ -50,15 +51,22 @@ public class ParquetMetricsRowGroupFilter {
 
   private final Schema schema;
   private final Expression expr;
+  private final Set<Integer> constantFieldIds;
 
   public ParquetMetricsRowGroupFilter(Schema schema, Expression unbound) {
-    this(schema, unbound, true);
+    this(schema, unbound, true, ImmutableSet.of());
```

Comment:
Without the extra check, any column whose stats are null go to ROWS_CANNOT_MATCH.  Thus any deletes table constant column filter at all will lead to 0 results.

Reply:
The fix here seems correct. I think we also have this issue in CDC. Let me check there.